### DISCUSSION TITLE
[design decision] Support comments or not

### DISCUSSION THREAD

**Discussion by josdejong** [tabular-json/tabular-json#1] [Ideas]
One very common complaint about JSON is that it doesn't support comments. Comments are particulary handy in configuration files. Tabular-JSON is not aiming to be used for configuration files, but since comments are quite easy to support and are a main complaint of JSON, it may be worth supporting them in Tabular-JSON.

In general, there are two types of comments:

```
// single line comments

/* multi
     line
       comments */
```

Before deciding on this, it will be good to do an experiment, implementing support for comments in a separate branch to see how this impacts the specification and implementation of a parser.

Complexity: Low
Added value: Medium

**Comment by gwhitney** [comment]
I strongly recommend allowing comments. You never know what a new format will ultimately be used for, and as Tabular-JSON is a human-sensible format, comment will always be helpful

**Reply by josdejong** [reply]
Thanks for your input Glen. You're right: the same holds for JSON itself, which is a data format but is often used as configuration format.

**Reply by gwhitney** [reply]
Yes, and the lack of comments is perpetually frustrating in projects that use JSON for configuration. In fact, that's why I submitted comment-handling code to `pnpm` for the case in which the package file is provided in JSON5 format, which does allow comments.

**Reply by josdejong** [reply]
😂 Comment noted.

I didn't know you where involved in getting comments working in `pnpm`, nice!

**Comment by josdejong** [comment]
Ok let's go for it and add support for comments. See #12.
